subsys/mgmt/mcumgr/smp_udp.c

```diff
@@ -204,7 +206,7 @@ int smp_udp_open(void)
 	memset(&addr4, 0, sizeof(addr4));
 	addr4.sin_family = AF_INET;
 	addr4.sin_port = htons(CONFIG_MCUMGR_SMP_UDP_PORT);
-	inet_pton(AF_INET, INADDR_ANY, &addr4.sin_addr);
+	addr4.sin_addr.s_addr = htonl(INADDR_ANY);
```

Comment:
The change does not seem related to claimed fix, should this receive separate commit in this PR?

Reply:
Hmmm... It _is_ a `NULL`-pointer dereference that is being fixed. `INADDR_ANY` (which is `0`) was being passed in place of a `char *` argument in `inet_pton()`, which resulted in a `NULL`-pointer dereference in `strlen()` (though the code would "silently" work in most systems which have a `0` at or near address `0`). While the fix itself may change the "API" being used, it is nevertheless in the spirit of the `NULL`-pointer dereference fixes that the PR describes.

Nevertheless, if you think it absolutely must be split out, please let me know and I can split the PR into two _really_ tiny ones.

Reply:
OK, my bad.

Reply:
No worries and thank you for the review.